apps/trading/components/banner/market-successor-banner.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { isBefore, formatDuration, intervalToDuration } from 'date-fns';
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import {
+  calcCandleVolume,
+  marketProvider,
+  useCandles,
+} from '@vegaprotocol/markets';
+import {
+  ExternalLink,
+  Intent,
+  NotificationBanner,
+} from '@vegaprotocol/ui-toolkit';
+import {
+  addDecimalsFormatNumber,
+  getMarketExpiryDate,
+  isNumeric,
+} from '@vegaprotocol/utils';
+import { t } from '@vegaprotocol/i18n';
+import { useLocalStorageSnapshot } from '@vegaprotocol/react-helpers';
+import * as Types from '@vegaprotocol/types';
+
+const getExpiryDate = (tags: string[], close?: string): Date | null => {
+  const expiryDate = getMarketExpiryDate(tags);
+  return expiryDate || (close && new Date(close)) || null;
+};
+
+export const MarketSuccessorBanner = () => {
+  const { pathname } = useLocation();
+  const isMarketPage = pathname.match(/^\/markets\/(?!(all)).+/);
+  const marketId = isMarketPage ? (pathname.split('/').pop() as string) : '';
+
+  const [dismissed, setDismissedInStorage] = useLocalStorageSnapshot(
```

Comment:
When did we agree that the banner should be dismiss-able and for that state to be stored? I find this unnecessary. If a market has been succeeded then in almost all cases the market will have terminated. Can we remove this functionality?

Reply:
I'm wrong on "all cases the market will have terminated". This is not true, there can be overlap between the parent and the successor. Nonetheless I'd like to not make the banner dismissable for this first iteration

Reply:
I removed functionality of storing the state